### ISSUE TITLE
Builder pattern

### ISSUE THREAD

**Issue by jgoldhammer** [gmessner/gitlab4j-api#235]
What do you think about redesigning the builder pattern in the current code base?
I would like to add a builder class to each model class so that this is not mixed anymore.

Example:

```
package org.gitlab4j.api.models;

import java.util.Date;

public final class SystemHookBuilder {
    private Integer id;
    private String url;
    private Date createdAt;
    private Boolean pushEvents;
    private Boolean tagPushEvents;
    private Boolean enableSslVerification;
    private Boolean repositoryUpdateEvents;

    public SystemHookBuilder id(Integer id) {
        this.id = id;
        return this;
    }

    public SystemHookBuilder url(String url) {
        this.url = url;
        return this;
    }

    public SystemHookBuilder createdAt(Date createdAt) {
        this.createdAt = createdAt;
        return this;
    }

    public SystemHookBuilder pushEvents(Boolean pushEvents) {
        this.pushEvents = pushEvents;
        return this;
    }

    public SystemHookBuilder tagPushEvents(Boolean tagPushEvents) {
        this.tagPushEvents = tagPushEvents;
        return this;
    }

    public SystemHookBuilder enableSslVerification(Boolean enableSslVerification) {
        this.enableSslVerification = enableSslVerification;
        return this;
    }

    public SystemHookBuilder repositoryUpdateEvents(Boolean repositoryUpdateEvents) {
        this.repositoryUpdateEvents = repositoryUpdateEvents;
        return this;
    }

    public SystemHook build() {
        SystemHook systemHook = new SystemHook();
        systemHook.setId(id);
        systemHook.setUrl(url);
        systemHook.setCreatedAt(createdAt);
        systemHook.setPushEvents(pushEvents);
        systemHook.setTagPushEvents(tagPushEvents);
        systemHook.setEnableSslVerification(enableSslVerification);
        systemHook.setRepositoryUpdateEvents(repositoryUpdateEvents);
        return systemHook;
    }

    @Override
    public String toString() {
      SystemHook systemHook =  new SystemHookBuilder().id().createdAt().build();
    }
}

```

A call to create a system hook would be like this:
```
SystemHook systemHook =  new SystemHookBuilder().id("1234").url("https://gitlab.com/webhook").build();
```

What do you think @gmessner ?

**Comment by gmessner** [comment]
This library already uses the fluent withFoo() pattern in a lot of model classes, in fact there was a contribution yesterday to add it to Group, Member, and Namspace (https://github.com/gmessner/gitlab4j-api/pull/231).

Using the withFoo() pattern would look similar to the builder pattern:
```
SystemHook systemHook =  new SystemHook().withId(1234).withUrl("https://gitlab.com/webhook");
```
Currently the withFoo() methods are implemented in the following org.gitlab4j.api.models classes:
```
CommitAction
Epic
Group
Member
Namespace
Project
PushRules
User
```

**Comment by gmessner** [comment]
@jgoldhammer 
I've released version 4.8.42 which includes the withFoo() methods in many of the model classes, including SystemHook.  I also added a static ```toJsonString(```) method to ```org.gitlab4j.api.utils.JacksonJson``` that dumps the object in JSON format with the class property names.   As model classes are modified I will be using this to add toString() methods to most.
